### PR TITLE
Update reason example

### PR DESCRIPTION
**Pull request by ulrikstrid**
Following up on #512

### CODE REVIEW

File: examples/with-reason-react/src/server.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ server
   .use(express.static(process.env.RAZZLE_PUBLIC_DIR))
   .get('/*', (req, res) => {
     const markup = renderToString(
-      <App title="Welcome to Razzle Reason React" />
+      <App title="Welcome to Razzle Reason React" initialUrl={req.url.substring(1)} />
```
Comment (ulrikstrid) [inline examples/with-reason-react/src/server.js]: This is kind of a hack to get rid of the initial "/", should I change it to something else?

File: examples/with-reason-react/src/server.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import express from 'express';
 import { renderToString } from 'react-dom/server';
 
-const App = require('../lib/js/src/app').comp; // BuckleScript output directory
+import App from '../lib/js/src/app'; // BuckleScript output directory
```
Comment (ulrikstrid) [inline examples/with-reason-react/src/server.js]: We can change `"in-source"` in  bsconfig.json to `true` and just require it like normal, `import App from './App'`

**Comment by ulrikstrid** [comment]
The Router and Link components are borrowed from https://github.com/thangngoc89/reason-react-router with slight modifications to allow passing in a URL for ssr.

File: examples/with-reason-react/src/client.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ import './client.css';
 import React from 'react';
 import { render } from 'react-dom';
 
-const App = require('../lib/js/src/app').comp; // BuckleScript output directory
+import App from '../lib/js/src/App'; // BuckleScript output directory
```
Comment (ulrikstrid) [inline examples/with-reason-react/src/client.js:6]: We can change "in-source" in bsconfig.json to true and just require it like normal, `import App from './App'`

**Comment by jaredpalmer** [comment]
> The Router and Link components are borrowed from https://github.com/thangngoc89/reason-react-router with slight modifications to allow passing in a URL for ssr.

Does the built-in reason router support SSR?

**Comment by ulrikstrid** [comment]
If you take a look at the Router component there is nothing that needs the browser there. The only thing that is needed is for us to pass a initial URL because then we will use that instead of dangerously getting the URL.
So on the server we really don't use the router, we just use the switch statement in App.re.

**Comment by drejohnson** [comment]
I've just noticed that going to root url actually renders "page not found" instead of home component. Going to "/home" or "/counter" rendered correctly on the server.

**Comment by ulrikstrid** [comment]
Noticed that now as well. In my app I added this util:
```ocaml
/* copied from  ReasonReact */
let urlToUrlList = url =>
  switch url {
  | ""
  | "/" => []
  | _ =>
    /* remove the preceeding /, which every pathname seems to have */
    let raw = Js.String.sliceToEnd(~from=1, url);
    /* remove the trailing /, which some pathnames might have. Ugh */
    let raw =
      switch (Js.String.get(raw, Js.String.length(raw) - 1)) {
      | "/" => Js.String.slice(~from=0, ~to_=-1, raw)
      | _ => raw
      };
    raw |> Js.String.split("/") |> Array.to_list;
  };
```
And then parse the URL with that (this also solved that I had to remove the leading "/"). Not sure if that is too much to add for this simple example or if I should include it. Would love for  ReasonReact to expose it instead.

**Comment by drejohnson** [comment]
I would add this to the example for now but it would be preferable if ReasonReact exposed something like this.

**Comment by ulrikstrid** [comment]
@jaredpalmer I think this is fine. Anything else you want me to fix?

**Pull request merged**